app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -464,11 +463,17 @@ export default Component.extend(TextareaTextManipulation, {
         this.topicId,
         this.categoryId,
         this.siteSettings.max_oneboxes_per_post,
-        false,
-        true
+        /* refresh */ false,
+        /* offline */ true
       );
 
-      resolveCachedShortUrls(this.siteSettings, cookedElement);
+      resolveAllShortUrls(ajax, this.siteSettings, cookedElement);
+
+      // trigger all the "api.decorateCookedElement"
+      this.appEvents.trigger(
```

Comment:
This is the **main** fix of this PR. This ensure we trigger all the plugins that use the `decorateCookedElement` to modify the preview after it's been cooked.

Reply:
I wonder if that can cause any perf issue if some plugins are doing heavy work here

Reply:
Where did this trigger get moved from? I don't see any relevant removed lines in this diff 🤔 

Or ... are we running it twice now?

Reply:
Yeah, I'm working on making it run only once (before when morphing, after when not)